pkg/secretless/plugin/sharedobj/available_plugins.go

```diff
@@ -105,3 +97,27 @@ func AllAvailablePluginsWithOptions(
 		TCPPluginsByID:  tcpPlugins,
 	}, nil
 }
+
+// NewEmptyPlugins plugins creates a new instance of Plugins with both maps
+// initialized but empty.
+func NewEmptyPlugins() Plugins {
```

Comment:
Why not just `NewPlugins` - It seems like one can assume that they are empty (and not `nil` like they would be if you instantiate `Plugins` struct directly.